content/en/service_management/app_builder/expressions.md

```diff
@@ -0,0 +1,121 @@
+
+### Conditional visibility
+
+You can make the visibility of a component dependent on other components.
+
+For example, if you want a text component to be visible only when two text input components named `textInput0` and `textInput1` have values, use the expression `${textInput0.value && textInput1.value}` in the **Is Visible** property.
+
+### Disable a component conditionally
+
+Similar to visibility, you can disable a component unless conditions are met by other components in your app.
+
+For example, if your app has a button that uses the content from a text component to send a message, you can disable the button unless the text component is visible:
+1. Click the button component on your canvas.
+1. Click the code editor symbol (**</>**) next to the **Is Disabled** property.
+1. Add the expression `${!text0.isVisible}`.
+
+The text component is invisible and the button is disabled unless both text input fields have content.
+
+{{< img src="service_management/app_builder/is-disabled.png" alt="The text component is invisible and the button is disabled unless both text input fields have content." style="width:100%;" >}}
+
+### Disable a component while loading
+
+Another common use case is disabling a component while a query is in a loading state. In the [EC2 Instance Manager blueprint][3], the `instanceType` Select component is disabled while the `listInstances` query is loading. To accomplish this, the **Is Disabled** property uses the expression `${listInstances.isLoading}`.
+
+{{< img src="service_management/app_builder/isloading.png" alt="The 'instanceType' Select component is disabled while the 'listInstances' query is loading." style="width:100%;" >}}
+
+## Custom query interactions
+
+Similar to components, you can use JS expressions to alter your queries based on user interaction.
+
+### Filter query results on user input
+
+The [PagerDuty On-call Manager blueprint][4] filters the result of the `listSchedules` query based on input from the user. The user selects a team and user from the `team` and `user` **Select** components.
+
+Inside the `listSchedules` query, the following post-query transformation filters the results based on the values of `team` and `user`:
+
+{{< code-block lang="js" disable_copy="true" >}}
+return outputs.body.schedules.map( s => {
+    return {
+        ...s,
+        users: s.users.map(u => u.summary),
+        teams: s.teams.map(u => u.summary)
+    }
+}).filter(s => {
+
+        const matchesName = !name.value.length ? true : s.name.toLowerCase().includes(name.value.toLowerCase());
+        const matchesTeam = team.value === 'Any Team' ? true : s.teams.includes(team.value);
+        const matchesUser = user.value === 'Any User' ? true : s.users.includes(user.value);
+
+        return matchesName && matchesUser && matchesTeam ;
+    }) || []
```

Comment:
I'm just curious what the `|| []` bit at the end means/does

Reply:
I believe it assigns an empty array if the previous expression returns a falsy/NaN etc.

Reply:
Neat, thanks